### ISSUE TITLE
Bits Support

### ISSUE THREAD

**Issue by Colbydude** [Colbydude/Music-Requests-Twitch-Extension#15]
Add support for Bits in the extension by adding request boosting and an option to only allow requests to be submitted using bits.

**Comment by ICTman1076** [comment]
Thanks for making this extension. One thing about bits support: please allow the streamer to set their own value. While some streamers might charge 100 bits, I don't want to do that as a small streamer as it seems unfair on my viewers. Preferably I want to be able to totally customise it, but even if it only goes up in increments of 5 bits it'd be far better than most extensions.

Thanks again :)

**Comment by Colbydude** [comment]
Hi ICTman!

This is a great suggestion! Something I believe that's also a Twitch Extensions best practice. I'll definitely set it up this way! :)

**Comment by ICTman1076** [comment]
Thank you, it's just annoying when some *glares at pretzelrocks* forces you to charge 100 bits lol

**Comment by Salvatos** [comment]
Wouldn’t this be against Twitch ToS regarding using bits to purchase/sell services or goods?

**Comment by ICTman1076** [comment]
> Wouldn’t this be against Twitch ToS regarding using bits to purchase/sell services or goods?

If it were why would they give extension developers a 20% cut?

**Comment by Salvatos** [comment]
> > Wouldn’t this be against Twitch ToS regarding using bits to purchase/sell services or goods?
> 
> If it were why would they give extension developers a 20% cut?

Odd reasoning. Just because extensions can make use of bits doesn’t imply that anything developers or streamers can possibly do with them is allowed. Streamers get a cut from bits, but that doesn’t mean they can do whatever they want with them. The same goes with extensions.

I’m thinking specifically of these examples of things not to do in their terms:
- _Don’t provide items, or specific services, that are associated with a monetary value in exchange for Bits._
- _Requesting a specific pre-recorded song to be played in exchange for an amount of Bits._

Whether the second example applies equally to live performances is ambiguous to me, but it seems arguable that a live song performance has a monetary value, especially for streamers who restrict song requests to donors.

Either way, I don’t have a definitive answer, I’m just saying Colbydude should check with Twitch Legal before potentially developing a feature that could get the extension or streamers banned.

**Comment by Colbydude** [comment]
> Either way, I don’t have a definitive answer, I’m just saying Colbydude should check with Twitch Legal before potentially developing a feature that could get the extension or streamers banned.

Yes there is a bit of a gray area on how this _could_ be supported, which is why it's far down on the backlog. However, no need to worry as the extension wouldn't even make it past the approval process before anyone would get banned.

**Comment by ICTman1076** [comment]
> Odd reasoning. Just because extensions can make use of bits doesn’t imply that anything developers or streamers can possibly do with them is allowed.

I'm more looking at the point of view that Twitch wouldn't want to promote anti-ToS use.

>     * Requesting a specific pre-recorded song to be played in exchange for an amount of Bits.

Well that's strange since extensions that do exactly that exist - Twitch has even partnered with one of them iirc for the music beta - so either this is new or unenforced.
